core/src/main/java/fr/sncf/osrd/api/TimetableCacheManager.kt

```diff
@@ -171,4 +184,65 @@ class TimetableCacheManager(
             return map
         }
     }
+
+    /**
+     * Try to get the cached value from valkey, returns null if the value isn't cached or if an
+     * error happened.
+     */
+    private suspend fun tryGetFromValkey(
+        valkeyConnection: StatefulRedisConnection<String, String>,
+        key: String,
+    ): STDCMRequirements? {
+        try {
+            val async = valkeyConnection.async()
+            val data = async.get(key).await() ?: return null
+            logger.info("valkey cache hit")
+
+            val cbor = Cbor {}
+            val serializer = STDCMRequirements.SerializableMap.serializer()
+            val serializableMap = cbor.decodeFromHexString(serializer, data)
+            return serializableMap.toSTDCMRequirements()
+        } catch (e: Exception) {
+            logger.info("error when fetching valkey cache: ${e.message}")
+            return null
+        }
+    }
+
+    /** Write the value to valkey, not blocking. */
+    private fun writeCacheToValkey(
+        valkeyConnection: StatefulRedisConnection<String, String>,
+        key: String,
+        data: STDCMRequirements,
+    ) {
+        val async = valkeyConnection.async()
+        val cbor = Cbor {}
+        val serializer = STDCMRequirements.SerializableMap.serializer()
+        val serialized = cbor.encodeToHexString(serializer, data.toSerializable())
+        val expirationMS = 24L * 60L * 60L * 1000L // One day
+        async.psetex(key, expirationMS, serialized).exceptionally {
+            logger.error("failed to send cached timetable to valkey: ", it)
```

Comment:
I guess the log level should match fetching exception:
```suggestion
            logger.warn("failed to send cached timetable to valkey: ", it)
```

Reply:
Fixed in 5dfa7dc6db7076ccaabd161ff3cbfe3a182f4b58